src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/ItemManager.cs

```diff
@@ -307,14 +307,30 @@ private static bool IsAfterDot(ITextSnapshot snapshot, ITrackingSpan applicableT
 
             var chosenItems = filterMethod(matchingItems, filterText);
 
-            var selectedItemIndex = 0;
+            int selectedItemIndex;
             VSCompletionItem uniqueItem = null;
             MatchResult bestOrFirstMatchResult;
 
             if (chosenItems.Length == 0)
             {
-                // We do not have matches: pick the first item from the list.
-                bestOrFirstMatchResult = itemsInList.FirstOrDefault();
+                // We do not have matches: pick the one with longest common prefix or the first item from the list.
+                selectedItemIndex = 0;
+                bestOrFirstMatchResult = itemsInList[0];
```

Comment:
do we know at thsi point that itemsInList is non-empty?

Reply:
yes, it was checked in the caller